pkg/secrets/fetch_secret.go

```diff
@@ -77,15 +97,13 @@ func execCommand(inputPayload string) ([]byte, error) {
 		var e *exec.ExitError
 		if errors.As(err, &e) {
 			exitCode = strconv.Itoa(e.ExitCode())
-		} else if ctx.Err() == context.DeadlineExceeded {
-			exitCode = "timeout"
```

Comment:
the `exitCode` is used as a tag value for telemetry, I think it's useful to keep `timeout` as a possible value, any reason to remove this?